plugins/module_utils/core.py

```diff
@@ -286,7 +301,8 @@ def present(
 
     def create_resource(self, type_name: str, params: Dict) -> bool:
         changed: bool = False
-        params = json.dumps(params)
+
+        params = json.dumps(params, cls=QuoteSwappingEncoder)
```

Comment:
This creates another problem. Consider the following task:

```yaml
    - amazon.cloud.iam_role:
        assume_role_policy_document:
          Version: 2012-10-17
          Statement:
            - Effect: Deny
              Principal:
                AWS: "*"
              Action: sts:AssumeRole
        description: "mgraves's awesome role"
        role_name: mgraves-awesome-role
```
Any string field that allows you to use single or double quotes is going to swap them, whether it makes any sense or not. I feel like we are in a vicious cycle where each change you make causes a new bug. I would suggest taking a step back and thinking about what problem it is you are trying to solve. It sounds like the point of the quote swapping and the  `merge_dicts()` logic is to merge IAM policy documents. If that's the case, then create a set of changes that is focused on doing only that. You won't have to deal with so many corner cases. The problem now is that you are trying to create something that can work on any dictionary, which is going to be very difficult to do correctly. It may even be impossible since there may be different behaviors needed for different resource properties.

Reply:
`merge_dicts()` is needed for the reason I explained before. It's not only tied to iam_role. The question of swapping quotes, I can isolate it to only a Policy parameter with a PolicyDocument suboption.

Reply:
> merge_dicts() is needed for the reason I explained before. It's not only tied to iam_role.

But the reason you gave was specifically tied to policies. Are there other cases?